### PR TITLE
Unlock Elevate Widget in Batch Gift Entry 

### PR DESCRIPTION
**Pull request by psadie-sfdo**
# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: src/lwc/geFormField/__mocks__/geFormField.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { LightningElement, api } from 'lwc';
```
Comment (psadie-sfdo) [inline src/lwc/geFormField/__mocks__/geFormField.js:1]: @bdvorachek - added a local mock and would love to get your early thoughts here.

File: .forceignore [hunk 2]
```diff
@@ -3,4 +3,5 @@
 **/.eslintrc.json
 
 # LWC Jest
-**/__tests__/**
\ No newline at end of file
+**/__tests__/**
+**/__mocks__/**
```
Comment (bdvorachek) [inline .forceignore:7]: 👍

File: src/lwc/geFormField/__mocks__/geFormField.js (hunk 1 above)
Comment (bdvorachek) [reply src/lwc/geFormField/__mocks__/geFormField.js:1]: I don't think we'd need to mock geFormField here, this folder would be for mocks that are specific to this component's tests. We could move the mock implementation of geFormService from geFormService.test.js into this folder as its own file and use it in the test.

File: src/lwc/geFormWidgetRowAllocation/geFormWidgetRowAllocation.html [hunk 3]
```diff
@@ -6,7 +6,7 @@
                         <div class="slds-col slds-size_1-of-3">
                             <lightning-input-field field-name={gauFieldApiName}
                                                    data-fieldname={gauFieldApiName}
-                                                   required='true'
+                                                   required
```
Comment (bdvorachek) [inline src/lwc/geFormWidgetRowAllocation/geFormWidgetRowAllocation.html:9]: 👍

File: src/lwc/geFormWidgetTokenizeCard/__mocks__/geFormWidgetTokenizeCard.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { LightningElement, api } from 'lwc';
```
Comment (bdvorachek) [inline src/lwc/geFormWidgetTokenizeCard/__mocks__/geFormWidgetTokenizeCard.js:1]: Similar here, we shouldn't need a mock for geFormWidgetTokenizeCard in this spot. This mock could get used in other tests that depend on this component.

File: src/lwc/geFormSection/__tests__/geFormSection.test.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+        it('should render 3 form fields', async () => {
+            const element = create(section);
+
+            await flushPromises();
+            const formFieldElements = shadowSelectorAll(element, 'c-ge-form-field');
+            expect(formFieldElements.length).toBe(3);
+        });
+
+        it('should render 3 form fields and the elevate widget', async () => {
```
Comment (bdvorachek) [inline src/lwc/geFormSection/__tests__/geFormSection.test.js:36]: One thing I'd like to see in this block of tests is one where we listen for the "registerPaymentWidgetEvent". A negative case where we assert that the event isn't fired, and a positive case where it does.

File: src/lwc/geFormSection/__tests__/geFormSection.test.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
+import { createElement } from 'lwc';
+import GeFormSection from 'c/geFormSection';
+
+import section from './data/section.json';
+import sectionWithElevateWidget from './data/sectionWithElevateWidget.json';
+
```
Comment (bdvorachek) [inline src/lwc/geFormSection/__tests__/geFormSection.test.js:6]:
```suggestion
const mockRegisterPaymentWidgetHandler = jest.fn();
```

File: src/lwc/geFormSection/__tests__/geFormSection.test.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+const shadowSelector = (element, selector) => {
+    return element.shadowRoot.querySelector(selector);
+}
+
+const create = (section) => {
+    const element = createElement('c-ge-form-section', {
+        is: GeFormSection
+    });
+    element.section = section;
```
Comment (bdvorachek) [inline src/lwc/geFormSection/__tests__/geFormSection.test.js:62]:
```suggestion
    element.section = section;
    element.addEventListener('registerpaymentwidget', mockRegisterPaymentWidgetHandler);
```

File: src/lwc/geFormSection/__tests__/geFormSection.test.js (hunk 5 above)
Comment (psadie-sfdo) [reply src/lwc/geFormSection/__tests__/geFormSection.test.js:36]: Great suggestion! Can do!

**Pull request closed without merging**